arcgis-ios-sdk-samples/Maps/Generate offline map (overrides)/GenerateOfflineMapOverridesViewController.swift

```diff
@@ -0,0 +1,340 @@
+        let paramNavigationController = storyboard!.instantiateViewController(withIdentifier: "OfflineParametersNavigationController") as! UINavigationController
+        let paramController = paramNavigationController.viewControllers.first as! OfflineMapParameterOverridesViewController
+        paramController.parameterOverrides = parameterOverrides
+        paramController.map = mapView.map
+        
+        // set the completion handler
+        paramController.startJobHandler = {[weak self] (paramController) in
+            // start the job
+            self?.takeMapOffline()
+        }
+        paramController.cancelHandler = {[weak self] (paramController) in
+            // reset the UI
+            self?.resetUIForOfflineMapGeneration()
+        }
+        //display the parameters sheet
+        present(paramNavigationController, animated: true)
+    }
+    
+    func resetUIForOfflineMapGeneration(){
+        
+        // close and reset the progress view
+        progressParentView.isHidden = true
+        progressView.progress = 0
+        progressLabel.text = ""
+        
+        //enable take map offline bar button item
+        barButtonItem.isEnabled = true
+        //unhide the extent view
+        extentView.isHidden = false
+    }
+    
+    //MARK: - Actions
+    
+    @IBAction func generateOfflineMapAction() {
+        
+        guard let offlineMapTask = offlineMapTask else {
+            return
+        }
+        
+        //disable bar button item
+        barButtonItem.isEnabled = false
+        //hide the extent view
+        extentView.isHidden = true
+        
+        //show progress hud
+        SVProgressHUD.show(withStatus: "Getting default parameters")
+        
+        //get the area outlined by the extent view
+        let areaOfInterest = extentViewFrameToEnvelope()
+        
+        //default parameters for offline map task
+        offlineMapTask.defaultGenerateOfflineMapParameters(withAreaOfInterest: areaOfInterest) { [weak self] (parameters: AGSGenerateOfflineMapParameters?, error: Error?) in
+            
+            if let error = error {
+                SVProgressHUD.showError(withStatus: error.localizedDescription)
+                return
+            }
+            
+            guard let parameters = parameters,
+                let self = self else {
+                return
+            }
+            
+            //dismiss progress hud
+            SVProgressHUD.dismiss()
+            
+            //will need the parameters for creating the job later
+            self.parameters = parameters
+            
+            //build the parameter overrides object to be configured by the user
+            offlineMapTask.generateOfflineMapParameterOverrides(with: parameters, completion: {[weak self] (parameterOverrides, error) in
+                
+                if let error = error {
+                    SVProgressHUD.showError(withStatus: error.localizedDescription)
```

Comment:
We are trying to get away from using `SVProgressHUD` for showing errors. Could you please use a `UIAlertController` instead?

Reply:
With pleasure.